src/modules/sync/advancedDiff.ts

```diff
@@ -0,0 +1,543 @@
+import {
+  ContentTypeElements,
+  ContentTypeModels,
+  ContentTypeSnippetModels,
+  TaxonomyModels,
+} from "@kontent-ai/management-sdk";
+import chalk from "chalk";
+import { readFileSync, writeFileSync } from "fs";
+import open from "open";
+import { resolve } from "path";
+
+import { SyncParams } from "../../commands/diff/diff.js";
+import { logError, logInfo } from "../../log.js";
+import {
+  isCountLimitation,
+  isDefaultValue,
+  isDependsOn,
+  isExternalIdReference,
+  isExternalIdReferenceArray,
+  isMaximumTextLength,
+  isObjectReference,
+  isObjectReferenceArray,
+  isValidationRegex,
+} from "../../utils/typeguards.js";
+import { RequiredCodename } from "../../utils/types.js";
+import { DiffModel, DiffObject } from "./types/diffModel.js";
+import {
+  AddIntoPatchOperation,
+  MovePatchOperation,
+  PatchOperation,
+  RemovePatchOperation,
+  ReplacePatchOperation,
+  ReplacePatchOperationValue,
+} from "./types/patchOperation.js";
+
+type DiffData = DiffModel & SyncParams;
+type EntityPathRenderer = {
+  regex: RegExp;
+  render: (() => string) | ((match: string[]) => string);
+};
+type Operation = PatchOperation["op"];
+type TypeOrSnippet = ContentTypeModels.IAddContentTypeData | ContentTypeSnippetModels.IAddContentTypeSnippetData;
+type ElementOrTerm = ContentTypeElements.Element | TaxonomyModels.IAddTaxonomyRequestModel;
+type RenderFunction<T extends PatchOperation> = (patchOp: T) => string;
+
+export const writeDiffToFile = (diffData: DiffData) => {
+  const template = readFileSync(resolve("./", "diffTemplate.html"), "utf-8");
+  const resolvedTemplate = template.replace(
+    /{{.*?}}/g,
+    (match) => rendererMap.get(match)?.(diffData) ?? match,
+  );
+  const outputPath = resolve(
+    diffData.outPath!,
+    `diff_${new Date().toISOString().replace(/[:.-]/g, "_")}.html`,
+  );
+
+  try {
+    logInfo(
+      diffData,
```

Comment:
I would rather pass something like logOptions.. this is really hard to understand for someone who doesn't know the repo

Reply:
fixed